tests/src/Integration/Action/UserRoleAddTest.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Action\UserRoleAddTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Action;
+
+use Drupal\Tests\rules\Integration\RulesEntityIntegrationTestBase;
+use Drupal\Tests\rules\Integration\RulesUserIntegrationTestTrait;
+use Drupal\user\RoleInterface;
+
+/**
+ * @coversDefaultClass \Drupal\rules\Plugin\Action\UserRoleAdd
+ * @group rules_actions
+ */
+class UserRoleAddTest extends RulesEntityIntegrationTestBase {
+
+  use RulesUserIntegrationTestTrait;
+
+  /**
+   * The action that is being tested.
+   *
+   * @var \Drupal\rules\Core\RulesActionInterface
+   */
+  protected $action;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    $this->enableModule('user');
+    $this->action = $this->actionManager->createInstance('rules_user_role_add');
+  }
+
+  /**
+   * Tests the summary.
+   *
+   * @covers ::summary
+   */
+  public function testSummary() {
+    $this->assertEquals('Adds roles to a particular user', $this->action->summary());
+  }
+
+  /**
+   * Tests adding of one role to user. User should not be saved.
+   *
+   * @covers ::execute
+   */
+  public function testAddOneRoleNoSave() {
+    // Set-up a mock user.
+    $account = $this->getMockedUser();
+    $account->expects($this->once())
+      ->method('hasRole')
+      ->with($this->equalTo('administrator'))
+      ->will($this->returnValue(FALSE));
+    $account->expects($this->once())
+      ->method('addRole')
+      ->with($this->equalTo('administrator'));
+    // We do noe expect call of the 'save' method because user should be
+    // auto-saved later.
+    $account->expects($this->never())
+      ->method('save');
+
+    // Mock the 'administrator' user role.
+    $administrator = $this->getMockedUserRole('administrator');
+
+    // Test adding of one role.
+    $this->action
+      ->setContextValue('user', $account)
+      ->setContextValue('roles', [$administrator])
+      ->execute();
```

Comment:
After that we need to call autoSaveContext() on the plugin and make sure that it returns ['user'].